addons/website_mass_mailing/static/src/js/website_mass_mailing.js

```diff
@@ -91,7 +91,7 @@ publicWidget.registry.subscribe = publicWidget.Widget.extend({
     },
 
     _getListId: function () {
-        return this.$el.closest('[data-snippet=s_newsletter_block').data('list-id') || this.$el.data('list-id');
+        return this.$el[0].closest('section[data-list-id]')?.dataset.listId || this.$el[0].dataset.listId;
```

Comment:
- No need for "section"
- Use `this.el` instead of `this.$el[0]`
- You don't need the `|| ...` fallback: `closest` searches on the starting element itself first

Reply:
> No need for "section"

There's an issue if I remove `section` in the selector.

Since `s_newsletter_subscribe_form` also has this `data-list-id` attribute ([see here](https://github.com/odoo/odoo/pull/186281/files#diff-c3f88f14f3bbe72a8344bf02ea5dc7d0f4a2f155d207c175afd7564fea4a9814R31)), removing `section` in the selector will target the latter div when we need to target the section.

One solution would be to remove `data-list-id` in `s_newsletter_subscribe_form` but I don't know if it's used somewhere 🤔 .

What do you think?

Reply:
I don't know 🤔 , the initial newsletter snippet, `s_newsletter_block`, already worked like this: it has a `data-list-id` on the [section](https://github.com/odoo/odoo/blob/master/addons/website_mass_mailing/views/snippets_templates.xml#L44) and the [JS line](https://github.com/odoo/odoo/blob/master/addons/website_mass_mailing/static/src/js/website_mass_mailing.js#L94) targets this section using the `data-snippet` attribute (which is defined automatically I guess ?).

![Capture d’écran 2024-11-21 à 08 59 31](https://github.com/user-attachments/assets/f1c6759b-6192-4bf4-9610-a8a08c9d1d16)

This is why I used the same structure for this new snippet.

Reply:
EDIT: I had to change the line otherwise the newsletter popup snippet was broken. Apparently this one doesn't have a `data-list-id` 🤔 
The change : https://github.com/odoo/odoo/compare/f25d915fa54f9ba0a432e5df863f3ebfd4d98ad0..54f4cdcf6e159c24474ee1ce02c26a897a93cc3d